### PR TITLE
Update release schedule after 1.14

### PR DESCRIPTION
**Pull request by dsimansk**
- Update release schedule after 1.14

/cc @knative/technical-oversight-committee 
/cc @knative/knative-release-leads

### CODE REVIEW

File: mechanics/RELEASE-SCHEDULE.md [hunk 1]
```diff
@@ -22,21 +22,22 @@ With that said, it can be useful to have a list of when future releases will hap
 
 | Release | Date       | EOL        | Min K8s Version | Notes |
 | ------- | ---------- | ---------- | --------------- | ----- |
-| 1.14    | 2024-04-23 | 2024-10-29 | 1.28            |       |
-| 1.15    | 2024-07-23 | 2025-01-28 | -               |       |
-| 1.16    | 2024-10-22 | 2025-04-29 | 1.29            |       |
+| 1.15    | 2024-07-23 | 2025-01-21 | -               |       |
+| 1.16    | 2024-10-22 | 2025-04-22 | 1.29            |       |
+| 1.17    | 2025-01-21 | 2025-07-22 | -               |       |
```
Comment (creydr) [inline mechanics/RELEASE-SCHEDULE.md:27]:
```suggestion
| 1.15    | 2024-07-23 | 2025-01-21 | 1.28            |       |
| 1.16    | 2024-10-22 | 2025-04-22 | 1.29            |       |
| 1.17    | 2025-01-21 | 2025-07-22 | 1.29            |       |
```
nit. I would state the versions (or don't we know for sure yet?)

**Review by creydr — APPROVED** [review]
Only one nit. But LGTM in general.
/lgtm

/hold
in case you want to address it

File: mechanics/RELEASE-SCHEDULE.md (hunk 1 above)
Comment (dsimansk) [reply mechanics/RELEASE-SCHEDULE.md:27]: I wasn't sure yet, depends on K8s schedule. We might not have newer version for 1.15. I'll try to figure it out though.

Comment (dsimansk) [reply mechanics/RELEASE-SCHEDULE.md:27]: Per you suggestion, it makes sense at a first glance.

**Comment by dsimansk** [comment]
@creydr thanks, I've merged your suggestion. So another lgtm will be needed.

**Review by creydr — COMMENTED** [review]
/lgtm

**Pull request merged**